### ISSUE TITLE
ReflectingConverter does not support getters that aren't backed by a field

### ISSUE THREAD

**Issue by GoogleCodeExporter** [eserating/siren4j#11]
```
I am building a POC to prove that I can support multiple output formats from 
one service with minimal changes.  I am using both Spring HATEOAS and Siren as 
my two test formats.  For all intents and purposes, the Resource objects that I 
create are identical EXCEPT that for HATEOAS I need to extend ResourceSupport 
and for Siren4j I need to extend BaseResource.

As a result, I could end up with a lot of code duplication.  So I thought I 
would extract the core object info into a delegate object.  Then each resource 
would implement the same interface and use the delegate as the backing object.  
Then the resources can have the same signature, but extend from two different 
objects.

The problem is that the ReflectingConverter calls 
ReflectionUtils.getExposedFieldInfo(clazz) which only processes getter methods 
if it can find the Field reference.  Further, the @Siren4JProperty annotation 
only supports use on FIELD where as the @Siren4jPropertyIgnore supports both 
FIELD and METHOD.

I think that if @Siren4JProperty supported METHOD as well, then the 
getExposedFieldInfo could be amplified to use the property as assigned to the 
method or field.  You'd need to establish a priority in case someone put the 
annotation in both places.

What steps will reproduce the problem?
1. Create a resource that extends BaseResource
2. This resource should have getters that are not backed by class fields
3. Attempt to RefelctingConverter.newInstance().toEntity(resource)

What is the expected output? What do you see instead?
I expect any non-ignored getter to be included in the property collection of 
the Entity object.  None of the getters that do not have a field are 
represented in the final Entity.

What version of the product are you using? On what operating system?
1.0.11
jdk1.7.0_51 on Win7

Please provide any additional information below.
Here is a ridiculously simplified example.  Like I mentioned above, the purpose 
would be to have another ErrorResource impl that extended ResourceSupport from 
Spring HATEOAS.

public interface IErrorResource {
    Integer getHttpStatusCode();
    void setHttpStatusCode(Integer httpStatusCode);
}

public class BaseErrorResource implements IErrorResource {
    private Integer httpStatusCode;
    @Override
    public Integer getHttpStatusCode() {
        return httpStatusCode;
    }
    @Override
    public void setHttpStatusCode(final Integer httpStatusCode) {
        this.httpStatusCode = httpStatusCode;
    }
}

@Siren4JInclude(Include.NON_EMPTY)
@Siren4JEntity(name = "error")
public class ErrorResource extends BaseResource implements IErrorResource {
    private final IErrorResource delegate = new BaseErrorResource();

    @Override
    public Integer getHttpStatusCode() {
        return delegate.getHttpStatusCode();
    }
    @Override
    public void setHttpStatusCode(final Integer httpStatusCode) {
        delegate.setHttpStatusCode(httpStatusCode);
    }
}

```

Original issue reported on code.google.com by `chuckdea...@gmail.com` on 17 May 2014 at 12:30

**Comment by zstd** [comment]
+ 1
Currently I am making proof of concept for using [immutables](https://github.com/immutables/immutables) as Siren resources. It works fine for simple types (Strings, numbers) that have no need to be annotated with Siren4JProperty, but for complex user-defined types I cannot use Siren4JProperty on methods and thus they are skipped from siren resource.

**Comment by zstd** [comment]
Implemented support for methods not backed by fields. See [PR](https://github.com/eserating/siren4j/pull/30).

**Comment by eserating** [comment]
Will be fixed in version 2.0
